presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -83,7 +83,13 @@ PrestoExchangeSource::PrestoExchangeSource(
       clientCertAndKeyPath_(clientCertAndKeyPath),
       ciphers_(ciphers),
       exchangeExecutor_(executor) {
-  folly::SocketAddress address(folly::IPAddress(host_), port_);
+  folly::SocketAddress address;
```

Comment:
@nmahadevuni Thanks for looking in this. I tried this and still seeing from here:

https://github.com/facebook/folly/blob/main/folly/SocketAddress.cpp#L634-L639

We will need to keep the code path where an IP address object is directly specified. For hostname, lets create a different constructor. Basically when node.ip is provided, we don't need to verify if its a host or ip, it will use the constructor which takes IPAdress. When node.internal-address is provided, it can be host or IP. Then we can construct PrestoExchange like you did earlier.